prometheus_juju_backup_all_exporter/exporter.py

```diff
@@ -56,18 +46,11 @@ def register(self, collector: Collector) -> None:
 
     def run(self, daemon: bool = False) -> None:
         """Start the exporter server."""
-        addr_family, addr = _get_best_family(self.addr, self.port)
-
-        class TmpServer(ThreadingWSGIServer):
-            """Copy of the ThreadingWSGIServer."""
-
-            address_family = addr_family
-
         httpd = make_server(
-            addr,
+            self.addr,
             self.port,
             self.app,
-            server_class=TmpServer,
+            server_class=ThreadingWSGIServer,
```

Comment:
This part I could not understand properly, can you please provide more information why we removed the _get_best-family function and TmpServer?

Reply:
Sorry, I was not being clear in the PR description. The reason we have `_get_best_family` in the first place is that we want to follow what was done in the [prom client](https://github.com/prometheus/client_python/blob/master/prometheus_client/exposition.py#L152) when creating a wsgi server, except that we want to make the wsgi server runs in non-daemon thread. The use of   `_get_best_family` in the [prom client](https://github.com/prometheus/client_python/blob/master/prometheus_client/exposition.py#L152) is to better support the server when binding to a IPv6 address. Since we will not change the `addr` parameter, it's not so useful in have `_get_best_family`

Reply:
Thanks for explanation :+1: